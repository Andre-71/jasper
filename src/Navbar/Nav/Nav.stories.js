import React from "react"
import { storiesOf } from "@storybook/react"

import Nav from "."

storiesOf("Nav").add("Nav", () => (
  <Nav>
    <Nav.Link href="#" active>Home</Nav.Link>
    <Nav.Link href="#">About</Nav.Link>
    <Nav.Link href="#">Seminar</Nav.Link>
    <Nav.Link href="#">Exhibitor</Nav.Link>
    <Nav.Link href="#">Location</Nav.Link>
  </Nav>
))
