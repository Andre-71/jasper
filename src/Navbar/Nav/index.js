import React from "react"

import PropTypes from "prop-types"
import styles from "./styles.module.css"
import findByType from "../../utils/findByType"

const Link = () => null

const Nav = ({ children }) => {
  const renderLink = () => {
    const links = findByType(children, Link)
    if (!links) return

    return links.map((link, idx) => (
      <li key={idx} className={`${styles.navItem} ${link.props.active ? styles.active : ""}`}>
        <a href={link.props.href}>
          {link.props.children}
        </a>
      </li>
    ))
  }

  return (
    <ul className={`${styles.nav}`}>
      {renderLink()}
    </ul>
  )
}

Nav.propTypes = {
  children: PropTypes.node
}

Nav.Link = Link
export default Nav
