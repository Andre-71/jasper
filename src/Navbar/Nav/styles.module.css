.nav {
  --accent: #44C3DB;

  display: flex;
}

.navItem {
  font-size: 14px;
  min-height: 48px;
  display: flex;
  opacity: 50%;
  font-family: "ITC Avant Garde Std Bk", Arial, Helvetica, sans-serif;
  line-height: 0px;
}

.navItem:hover {
  opacity: 80%;
}

.navItem:not(:last-child) {
  margin-right: 34px;
}

.navItem.active {
  font-weight: 600;
  opacity: 100%;
}

.navItem > * {
  flex: 1 1;
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

@media only screen and (min-width: 768px) {
  .navItem {
    font-size: 16px;
    position: relative;
  }

  .navItem:not(:last-child) {
    margin-right: 48px;
  }

  .navItem.active::before {
    content: " ";
    position: absolute;
    background: var(--accent);
    height: 4px;
    width: calc(100% + 14px);
    transform: translateX(-50%);
    left: 50%;
    bottom: 0;
  }
}

