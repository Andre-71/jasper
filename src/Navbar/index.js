import React, { useState } from "react"

import PropTypes from "prop-types"
import styles from "./styles.module.css"

import findByType from "../utils/findByType"
import useGetWindowWidth from "../utils/useGetWindowWidth"
import { BREAKPOINT } from "../utils/variables"

import ToggleClose from "./assets/toggle_close.svg"
import ToggleOpen from "./assets/toggle_open.svg"

const Brand = () => null
const Collapse = () => null
const Toggle = () => null

const Navbar = ({ children }) => {
  const [show, setShow] = useState(false)
  const windowWidth = useGetWindowWidth()
  const isMobile = windowWidth < BREAKPOINT

  const renderBrand = () => {
    const brand = findByType(children, Brand)[0]
    if (!brand) return

    let brandText, brandTextMobile
    const brandChild = brand.props?.children || ""

    if (!Array.isArray(brandChild)) {
      brandText = brandTextMobile = brandChild
    } else {
      brandTextMobile = brandChild.filter((b) => b.props.mobile)
      brandText = brandChild.filter((b) => !b.props.mobile)
    }

    const brandLink = brand.props?.href || "#"

    return (
      <div className={styles.brand}>
        <div className={styles.brandWrapper}>
          <a href={brandLink}>
            <div className={styles.brandText}>
              {isMobile ? brandTextMobile : brandText}
            </div>
          </a>
          {renderToggle()}
        </div>
      </div>
    )
  }

  const renderToggle = () => {
    const toggle = findByType(children, Toggle)[0]

    if (!toggle) return

    return (
      <div className={styles.toggle} onClick={() => setShow((prev) => !prev)}>
        {show ? <img src={ToggleClose} /> : <img src={ToggleOpen} />}
      </div>
    )
  }

  const renderCollapse = () => {
    const collapse = findByType(children, Collapse)[0]
    if (!collapse) return

    return (
      <div
        className={`${styles.collapse} ${show || !isMobile ? styles.show : ""}`}
      >
        {collapse.props.children}
      </div>
    )
  }

  return (
    <div className={styles.navbar}>
      <div className={styles.navbarWrapper}>
        {renderBrand()}
        {renderCollapse()}
      </div>
    </div>
  )
}

Navbar.propTypes = {
  children: PropTypes.node
}

Navbar.Brand = Brand
Navbar.Collapse = Collapse
Navbar.Toggle = Toggle
export default Navbar
