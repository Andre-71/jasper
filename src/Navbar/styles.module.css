* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.navbar {
  --bg: #080E16;
  --fg: #FFFFFF;

  background: var(--bg);
  color: var(--fg);
}

.navbarWrapper {
  padding: 0 1.4em;
}

.brandText {
  display: flex;
}

.brandText > *:not(:last-child) {
  margin-right: 8px;
}

.brand {
  display: flex;
  min-height: 48px;
}

.brandWrapper {
  display: flex;
  flex: 1;
  justify-content: space-between;
  align-items: center;
}

.brandWrapper > * {
  outline: none;
}

.brandText * {
  display: block;
}

.collapse {
  overflow: hidden;
  display: flex;
  max-height: 0;
  transition: max-height 0.4s cubic-bezier(0, 1, 0, 1);
}

.collapse > * {
  flex-direction: column;
  width: 100%;
  margin-bottom: 24px;
}

.collapse > * > * {
  margin: 0 12px;
}

.collapse.show {
  max-height: 1000px;
  transition: max-height 0.4s ease-in-out;
}

.toggle {
  cursor: pointer;
}

.toggle > * {
  display: block;
}

@media only screen and (min-width: 789px) {
  .collapse {
    overflow: initial;
  }

  .collapse > * {
    flex-direction: row;
    margin-bottom: 0;
  }

  .collapse > * > * {
    padding: 0;
  }

  .collapse > * > *:not(:last-child) {
    margin-right: 48px;
  }

  .toggle {
    display: none;
  }

  .navbarWrapper {
    display: flex;
    justify-content: space-between;
    padding: 0 4em;
  }

  .brand {
    min-height: 67px;
  }
}
