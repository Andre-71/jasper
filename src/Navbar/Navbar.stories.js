import React from "react"
import { storiesOf } from "@storybook/react"

import Navbar from "."
import Nav from "./Nav"
import { ReactComponent as LogoMobile } from './assets/logo_mobile.svg'
import { ReactComponent as LogoDesktop } from './assets/logo_desktop.svg'

storiesOf("Navbar").add("Navbar", () => (
  <Navbar>
    <Navbar.Brand>
      <div mobile>
        <LogoMobile/>
      </div>
      <div>
        <LogoDesktop/>
      </div>
    </Navbar.Brand>
    <Navbar.Toggle />
    <Navbar.Collapse>
      <Nav>
        <Nav.Link href="#" active>Home</Nav.Link>
        <Nav.Link href="#">About</Nav.Link>
        <Nav.Link href="#">Seminar</Nav.Link>
        <Nav.Link href="#">Exhibitor</Nav.Link>
        <Nav.Link href="#">Location</Nav.Link>
        <Nav.Link href="#">Playground</Nav.Link>
        <Nav.Link href="#">Competition</Nav.Link>
      </Nav>
    </Navbar.Collapse>
  </Navbar>
))
