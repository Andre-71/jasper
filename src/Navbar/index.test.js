import React from "react"
import renderer from "react-test-renderer"
import Navbar from '.'
import Nav from './Nav'

it("renders correctly", () => {
  const tree = renderer.create((
    <Navbar>
      <Navbar.Brand>
        <div>
          Hello
        </div>
      </Navbar.Brand>
      <Navbar.Toggle />
      <Navbar.Collapse>
        <Nav>
          <Nav.Link href="#" active>Budi</Nav.Link>
          <Nav.Link href="#">Andi</Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  )).toJSON()
  expect(tree).toMatchSnapshot()
})
