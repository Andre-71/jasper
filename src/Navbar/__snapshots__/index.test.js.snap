// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navbar"
>
  <div
    className="navbarWrapper"
  >
    <div
      className="brand"
    >
      <div
        className="brandWrapper"
      >
        <a
          href="#"
        >
          <div
            className="brandText"
          >
            <div>
              Hello
            </div>
          </div>
        </a>
        <div
          className="toggle"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="13"
            viewBox="0 0 26 13"
            width="26"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="white"
              height="4"
              rx="2"
              transform="matrix(1 0 0 -1 0 4)"
              width="26"
            />
            <rect
              fill="white"
              height="4"
              rx="2"
              transform="matrix(1 0 0 -1 0 13)"
              width="26"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="collapse show"
    >
      <ul
        className="nav"
      >
        <li
          className="navItem active"
        >
          <a
            href="#"
          >
            Budi
          </a>
        </li>
        <li
          className="navItem "
        >
          <a
            href="#"
          >
            Andi
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
